libs/tree-sitter-wing/grammar.js

```diff
@@ -160,6 +162,22 @@ module.exports = grammar({
         $._semicolon
       ),
 
+    variable_assignment_incr_statement: ($) =>
+      seq(
+        field("name", alias($.reference, $.lvalue)),
+        "+=",
+        field("value", $.expression),
```

Comment:
There's some duplication here - could we combine these? e.g.

```
    variable_assignment_statement: ($) =>
      seq(
        field("name", alias($.reference, $.lvalue)),
        field("operator", $.assignment_operator),
        field("value", $.expression),
        $._semicolon
      ),
   
   assignment_operator: _ => choice("=", "+=", "-="),
```

Reply:
Good catch, this `grammar.js` syntax is new to me, so I settled with a way that worked, which turned out not to be the best - yours looks much better. I will look into it later in the day, thank you :)

Reply:
I have tried to combine it, but it seems like it can not be translated properly this way. The compiler does not differentiate between assignments - it treats all - `=`, `+=`, `-=` -  as `=`, therefore tests fail.

I am trying to deep deeper and maybe find some work around.

Reply:
I modified `grammar.js` the same way as you suggested. In order for it to work I also had to modify [parser](https://github.com/winglang/wing/pull/4018/files#diff-f8d8b2a32f3fb730f7c3e76c059919266d79127c00c57c1eb98599f6ae284da1R441) and adjust [statements.txt](https://github.com/winglang/wing/pull/4018/files#diff-20471e3433dc0ea2e3af17c3568071e229b05d5eb15516a3deab142e64b351e6R91).

Reply:
Sweet